flat roster: pass master options to apply_sdb

jhass opened the pull request:
### What does this PR do?

Despite being more convenient and consistent, in avoiding to having to specify the sdb options
again for each host, `apply_sdb` (or actually `sdb_get` I think) also somehow manages to mess up
the previously passed options in the next iteration.

### Previous Behavior

Using sdb in the flat roster copied the entry of the current host into the key of the current one. It also required the sdb configuration to be present in each roster entry.

### New Behavior

Pass the master config available in `__opts__` to `apply_sdb`, centralizing the sdb configuration properly and keeping `sdb_get` from messing things up.

### Tests written?

No, I couldn't find any existing tests for this and writing an entire suite for this feature
is a bit beyond my experience with this project.

xeacott commented:
Hey @jhass thanks for submitting this PR! Unfortunately we won't be able to merge without tests for this. I realize there aren't any unit or integration tests, but maybe we can work together to get some written!

jhass commented:
Sure, any pointers?

xeacott commented:
So it looks like there aren't any tests for the flat file itself, however there are tests residing in tests/unit/roster/test_clustershell.py and i think that's a pretty good example of how to test flat.py :)

xeacott commented:
Also @jhass I would also ask if you could resolve the conflict please and also run pre-commit on this. If you are unsure of pre-commit I can run that for ya.

jhass commented:
Gladly take over, I have no clue what I'm doing, whether this is any sensible test. It feels like it's just asserting a particular implementation, which in my book is pretty useless. Also let's see what CI says, locally I just get

```
$ python tests/runtests.py --from-filenames=salt/roster/flat.py
[1]    90592 abort      python tests/runtests.py --from-filenames=salt/roster/flat.py 
```

jhass committed:
flat roster: pass master options to apply_sdb

Despite being more convenient and consistent, in avoiding to having to specify the sdb options
again for each host, apply_sdb (or actually sdb_get I think) also somehow manages to mess up
the previously passed options in the next iteration.

xeacott commented:
Thanks for getting that file together :) I'll get pre-commit ran for ya and then check out those tests! Sometimes they do seem really meaningless, especially when the change is _that_ minor but it helps increase our coverage!

dwoz commented:
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.
